### PR TITLE
Make Global Handlers optional in restfulWS-3.1 and xmlWS-4.0

### PR DESCRIPTION
**Pull request by WhiteCat22**
This PR removes GlobalHandlers from the EE10 web services features (`restfulWS-3.1` and `xmlWS-4.0`). Instead of being executed on every request, we only want to enable them if the user explicitly enables the `com.ibm.websphere.appserver.globalhandler-1.0` feature in their custom user feature.

https://github.com/OpenLiberty/open-liberty/issues/22598

### CODE REVIEW

**Comment by WhiteCat22** [comment]
#libby #build

**Comment by posmith** [comment]
L2 message review not required

**Comment by WhiteCat22** [comment]
#libby #build

**Comment by WhiteCat22** [comment]
#libby #build

**Comment by WhiteCat22** [comment]
#libby #build

**Comment by WhiteCat22** [comment]
#build

**Comment by WhiteCat22** [comment]
#build

**Comment by WhiteCat22** [comment]
#build

**Comment by WhiteCat22** [comment]
#build

**Comment by WhiteCat22** [comment]
#build

**Comment by WhiteCat22** [comment]
Build failed with a ConnectionTimeout
#Libby #Build

**Comment by WhiteCat22** [comment]
#build

**Comment by WhiteCat22** [comment]
#libby

**Review by neuwerk — DISMISSED** [review]
Changes look good to me!

File: dev/io.openliberty.jaxws.globalhandler.internal/src/com/ibm/ws/jaxws/globalhandler/GlobalhandlerSOAPMessageContextImpl.java [hunk 1]
```diff
@@ -10,6 +10,7 @@
  *******************************************************************************/
 package com.ibm.ws.jaxws.globalhandler;
 
+
```
Comment (jim-krueger) [inline dev/io.openliberty.jaxws.globalhandler.internal/src/com/ibm/ws/jaxws/globalhandler/GlobalhandlerSOAPMessageContextImpl.java:13]: This java file does not need to be included in this PR if all you are doing is adding spaces.

**Review by jim-krueger — DISMISSED** [review]
Approving with one minor comment.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.mpJwtPropagation-2.0.feature [hunk 2]
```diff
@@ -1,8 +1,8 @@
 -include= ~${workspace}/cnf/resources/bnd/feature.props
 symbolicName=io.openliberty.mpJwtPropagation-2.0
 visibility=private
-IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.restfulWSClient-3.0)(osgi.identity=io.openliberty.restfulWSClient-3.1)))", \
- osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.mpJwt-2.0)(osgi.identity=io.openliberty.mpJwt-2.1)))"
+IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.restfulWSClient-3.0)))", \
+ osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.mpJwt-2.0)))"
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.mpJwtPropagation-2.0.feature]:
```suggestion
 osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=io.openliberty.mpJwt-2.0))"
```

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.mpJwtPropagation-2.1.feature [hunk 3]
```diff
@@ -0,0 +1,13 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=io.openliberty.mpJwtPropagation-2.1
+visibility=private
+IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.restfulWSClient-3.1)))", \
+ osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.mpJwt-2.1)))"
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.mpJwtPropagation-2.1.feature]:
```suggestion
IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=io.openliberty.restfulWSClient-3.1))", \
 osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=io.openliberty.mpJwt-2.1))"
```

File: dev/io.openliberty.jaxws.globalhandler.internal/bnd.bnd [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+-include= ~../cnf/resources/bnd/bundle.props
+bVersion=1.0
+
+-sub: *.bnd
+
+Export-Package: \
+ com.ibm.ws.jaxws.globalhandler;version="4.0"
+
+Import-Package: \
+ com.ibm.ws.jaxws.globalhandler;version="2.3", \
```
Comment (jhanders34) [inline dev/io.openliberty.jaxws.globalhandler.internal/bnd.bnd]: split package?  You export it with one version and import it with another version.  Very odd.

File: dev/io.openliberty.jaxws.globalhandler.internal/.gitignore [hunk 5]
```diff
@@ -0,0 +1,2 @@
+dropins
+/bootstrapping.properties
```
Comment (jhanders34) [inline dev/io.openliberty.jaxws.globalhandler.internal/.gitignore]: This file makes no sense.  There should be no dropins or bootstrapping.properties for this project.  I think it should be deleted.

File: dev/io.openliberty.jaxws.globalhandler.internal/.classpath [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="src" path="fat/src"/>
```
Comment (jhanders34) [inline dev/io.openliberty.jaxws.globalhandler.internal/.classpath]:
```suggestion
```

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.xmlWS4.0-globalHandler1.0.feature [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.xmlWS-4.0)))", \
+ osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=com.ibm.websphere.appserver.globalhandler-1.0)))"
+-features\
+  io.openliberty.globalhandler1.0.internal.ee-10.0
+-bundles=\
+  io.openliberty.jaxws.globalhandler.internal.jakarta
+IBM-Install-Policy: when-satisfied
+kind=noship
+# noship features should have an edition of full instead of base
+edition=full
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.xmlWS4.0-globalHandler1.0.feature]:
```suggestion
kind=beta
edition=base
```

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.xmlWS4.0-globalHandler1.0.feature [hunk 8]
```diff
@@ -0,0 +1,14 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=io.openliberty.xmlWS4.0-globalHandler1.0
+visibility=private
+IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.xmlWS-4.0)))", \
+ osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=com.ibm.websphere.appserver.globalhandler-1.0)))"
+-features\
+  io.openliberty.globalhandler1.0.internal.ee-10.0
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.xmlWS4.0-globalHandler1.0.feature]:
```suggestion
IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=io.openliberty.xmlWS-4.0))", \
 osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=com.ibm.websphere.appserver.globalhandler-1.0))"
```

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restfulWS3.1-globalHandler1.0.feature [hunk 9]
```diff
@@ -0,0 +1,13 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=io.openliberty.restfulWS3.1-globalHandler1.0
+visibility=private
+IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.restfulWS-3.1)))", \
+ osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=com.ibm.websphere.appserver.globalhandler-1.0)))"
+-features=\
+  io.openliberty.globalhandler1.0.internal.ee-10.0
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restfulWS3.1-globalHandler1.0.feature]:
```suggestion
IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=io.openliberty.restfulWS-3.1))", \
 osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=io.openliberty.globalhandler-1.0))"
```

**Review by jhanders34 — CHANGES_REQUESTED** [review]
There is no way to have user feature use globalhandler since it isn't moved to be protected so this PR seems incomplete.  com.ibm.websphere.appserver.globalhandler-1.0 should be moved to be a protected feature.  I would actually prefer for it to be renamed to io.openliberty.globalhandler-1.0 when that happens.  I can't believe that no FATs fail as a result of this change.  Do we not have repeats for EE10 in the globalhandler tests?  I know that Berksan is working in EE10 repeats for XML WS 4.0.  Not sure if that will cause tests to start failing.  Are there RESTful WS 3.1 tests for global handler that need to be updated to do repeats?

Also `io.openliberty.globalhandler.spi; location:=dev/spi/ibm/` will need to be added to the `-jars` list in the protected feature and `dev/spi/ibm/javadoc/io.openliberty.globalhandler.spi_1.0-javadoc.zip` will need to be added the `-files` list in the protected feature.

File: dev/com.ibm.websphere.appserver.features/visibility/protected/io.openliberty.globalhandler-1.0.feature [hunk 10]
```diff
@@ -0,0 +1,9 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=io.openliberty.globalhandler-1.0
+visibility=protected
+-features=\
+  com.ibm.websphere.appserver.globalhandler-1.0, \
+  com.ibm.websphere.appserver.servlet-3.0; ibm.tolerates:="3.1,4.0,5.0,6.0"
+kind=ga
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/protected/io.openliberty.globalhandler-1.0.feature:7]:
```suggestion
kind=beta
```

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.xmlWS4.0-globalHandler1.0.feature [hunk 11]
```diff
@@ -0,0 +1,11 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=io.openliberty.xmlWS4.0-globalHandler1.0
+visibility=private
+IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=io.openliberty.restfulWS-3.1))", \
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.xmlWS4.0-globalHandler1.0.feature:4]:
```suggestion
IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=io.openliberty.xmlWS-4.0))", \
```

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.xmlWS4.0-globalHandler1.0.feature [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+ osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=com.ibm.websphere.appserver.globalhandler-1.0))"
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.xmlWS4.0-globalHandler1.0.feature:5]:
```suggestion
 osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=io.openliberty.globalhandler-1.0))"
```

**Comment by jhanders34** [comment]
The commits in this PR were merged in with PR #24040.  As such I am closing out this PR as well.

**Pull request closed without merging**